Assets/Creatures/api/abilities/ReactiveState.cs

```diff
@@ -0,0 +1,26 @@
+using Creatures.api.abilities;
+using Creatures.api.abilities.states;
+
+namespace Creatures.Api
+{
+    public abstract class ReactiveState : IAbilityState
+    {
+        protected AbilityEvent _abilityEvent;
+        
+        public abstract void Enter();
+        public abstract void Exit();
+        public abstract AbilityExecutionResult Execute(IPlayable playable, AbilityEvent abilityEvent);
+
+        public ReactiveState(AbilityEvent abilityEvent)
+        {
+            _abilityEvent = abilityEvent;
```

Comment:
could you use `this.` instead of the `_`?

Reply:
just a naming convention thing: https://learn.microsoft.com/en-us/dotnet/csharp/fundamentals/coding-style/identifier-names